tensorflow/contrib/data/python/ops/dataset_ops.py

```diff
@@ -887,6 +887,10 @@ def __init__(self, tensors):
     """See `Dataset.from_tensor_slices()` for details."""
     super(TensorSliceDataset, self).__init__()
     with ops.name_scope("tensors"):
+      tensors = ops.convert_n_to_tensor_or_indexed_slices(tensors)
```

Comment:
Hi @mrry,
I really like the idea of not having to specify `tf.constant(filenames)` and `tf.constant(labels)` when feeding them into the function to create a dataset.
Even in the `tf.contrib.data` README, you use as example:
```python
filenames = ["/var/data/file1.txt", "/var/data/file2.txt"]
dataset = tf.contrib.data.Dataset.from_tensor_slices(filenames)
```
And
```python
filenames = tf.constant(["/var/data/image1.jpg", "/var/data/image2.jpg", ...])
labels = tf.constant([0, 37, 29, 1, ...])

dataset = tf.contrib.data.Dataset.from_tensor_slices((filenames, labels))
```

Both examples don't work with the current version.

I think the easiest fix would be to force users to pass in a list, as in `tf.train.slice_input_producer`? We would have to rewrite some examples and kernel tests, but I feel it's more natural.